chia/util/keychain.py

```diff
@@ -561,8 +566,10 @@ def migrate_legacy_keyring(
 
     @staticmethod
     def get_keys_needing_migration() -> Tuple[List[Tuple[PrivateKey, bytes]], Optional["Keychain"]]:
-        legacy_keyring: Optional[Keychain] = Keychain(force_legacy=True)
-        if legacy_keyring is None:
+        try:
+            legacy_keyring: Keychain = Keychain(force_legacy=True)
+        except KeyringNotSet:
+            # No legacy keyring available, so no keys need to be migrated
```

Comment:
Catch the exception instead of attempting to check for `None` (which doesn't work)